src/main/java/frc5124/robot2020/commands/TargetTracker.java

```diff
@@ -0,0 +1,64 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc5124.robot2020.commands;
+
+import java.util.Set;
+
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import frc5124.robot2020.RobotMap;
+import frc5124.robot2020.subsystems.Turret;
+
+public class TargetTracker implements Command {
+
+    private final Turret turret;
+    private final PIDController turretPid;
+
+    public TargetTracker(Turret turret) {
+        this.turret = turret;
+        turretPid = new PIDController(RobotMap.Turret.P, RobotMap.Turret.I, RobotMap.Turret.D);
```

Comment:
best to set up the pid controller in the turret subsystem so it can be used by both manual turning and the tracking. pass a position/setpoint/reference point to the subsystem and let it run to position using its own closed loop